File: iteration2/Notification.java
package iteration2;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class Notification {
    private static final Logger logger = LogManager.getLogger(Notification.class);
    private String message;
    private Person recipient;


    public Notification(Person person, String message) {
        this.recipient = person;
        this.message = message;
    }

    public void sendNotification() {
        if (recipient instanceof Advisor) {
            ((Advisor) recipient).setNotification(this);
        } else if (recipient instanceof Student) {
            logger.info("Notification sent to student " + recipient.getID() + " that their request has been rejected.");
            ((Student) recipient).setNotification(this);
        }

    }

    public String getMessage() {
        return message;
    }
}

File: iteration3/Notification.py
import logging

from iteration3.Advisor import Advisor
from iteration3.Student import Student


class Notification:
    def __init__(self, recipient, message):
        self.recipient = recipient
        self.message = message

    def sendNotification(self):
        if isinstance(self.recipient, Advisor):
            self.recipient.setNotification(self)
        elif isinstance(self.recipient, Student):
            logging.info(f"Notification sent to student {self.recipient.getID()} that their request has been rejected.")
            self.recipient.setNotification(self)

    def getMessage(self):
        return self.message
